torch/csrc/lazy/ts_backend/ts_node_lowering.cpp

```diff
@@ -346,6 +356,37 @@ class TSNodeLowering : public TSNodeLoweringInterface {
     return LowerBuiltin(at::aten::reshape, arguments);
   }
 
+  TSOpVector LowerDiagonal(const Diagonal* node) {
+    std::vector<torch::jit::NamedValue> arguments;
+    arguments.emplace_back(loctx()->GetOutputOp(node->operand(0)));
+    arguments.emplace_back(node->offset());
+    arguments.emplace_back(node->dim1());
+    arguments.emplace_back(node->dim2());
+    return LowerBuiltin(node, arguments);
+  }
+
+  // FIXME(alanwaketan): One day we should code-gen all view ops, or at
+  // least move the lowering to the IR nodes.
+  TSOpVector LowerDiagonalViewUpdate(const DiagonalViewUpdate* node) {
```

Comment:
@Krovatkin What do you think about this approach to update the original tensor/view's values? Do you think it can be generalized, such as your `expand` case.?

Reply:
yeah, I think we can use this approach in general

Reply:
@bdhirsh Thought you might be interested in these.